engine/src/main/java/org/terasology/engine/module/ExternalApiWhitelist.java

```diff
@@ -28,9 +28,11 @@
             // TODO: This one org.terasology entry is a hack and needs a proper fix
             .add("org.terasology.math.geom")
             .add("java.lang")
+            .add("java.beans")
             .add("java.lang.invoke")
             .add("java.lang.ref")
             .add("java.math")
+            .add("java.nio")
```

Comment:
I always thought that IO was not whitelisted on purpose before, but I might be wrong. 

@jellysnake @eviltak anything to say about this?

Reply:
Not letting rogue mods delete your filesystem was one of the primary reasons for having a sandbox, yes.

Reply:
From Slack conversation with @immortius :

- we want to have some permission management, where the user is prompted to grant modules permissions to specific categories.
- The `@API` annotation has a parameter for specifying permission group, and module info does too: `@Api(TerasologyPermissionSets.FILE)`
- https://github.com/MovingBlocks/gestalt/blob/develop/gestalt-module/src/test/java/org/terasology/module/SandboxTest.java has examples
- Gestalt already uses permission groups to restrict what modules can access
- ideally nio would be in some sort of File permission group, and users would be prompted to permit the module given it may read/write/wipe files
- Where/how to store granted permissions (by user actions)? Probably just a list of approved modules, probably in config. Currently we don't have support for approving part of the permissions a module needs. It is the entire module that is approved